src/DotNetWorker.Core/StartupHook.cs

```diff
@@ -62,4 +70,22 @@ static bool WaitOnDebugger(int cycle)
             }
         }
     }
+
+    internal static void RemoveSelfFromStartupHooks()
+    {
+        string? startupHooks = Environment.GetEnvironmentVariable(StartupHooksEnvVar);
+        if (string.IsNullOrEmpty(startupHooks))
+        {
+            // If this call happened, we are clearly part of this environment variable.
+            // This is mostly to make strict-nulls happy.
+            return;
+        }
+
+        // netstandard2.0 has no StringSplitOptions overload.
+        IEnumerable<string> parts = startupHooks.Split(s_startupSeparator).Where(x => !string.IsNullOrEmpty(x));
+        string newValue = string.Join(
+            s_startupSeparator.ToString(), // netstandard2.0 only accepts string here.
+            parts.Where(x => !x.Equals(s_assemblyName, StringComparison.Ordinal)));
```

Comment:
Do we need to handle null or empty strings here? I'm inclined to say we shouldn't modify the input beyond removing our assembly. 

Wondering if we could simplify to something like:

``` csharp
            IEnumerable<string> parts = startupHooks.Split(s_startupSeparator[0])
                                                    .Where(x => !string.Equals(x, s_assemblyName, StringComparison.Ordinal));
            string newValue = string.Join(s_startupSeparator, parts);
```

The above relies on switching the separator from a `char` to a `string`, makes it a bit more concise/simpler and preserves all other input.